website/blog/2018-06-25-on-consuming-and-publishing-es6+-packages.md

```diff
@@ -0,0 +1,218 @@
+- React Native has always compiled `node_modules` by default and has probably learned a lot from issues like ^.
+- `preset-env` could have its own bugs because we use `compat-table` vs. test262.
+- Browsers themselves can have issues with running native ES6+ code vs. ES5.
+- Question of how do we determine what is "supported": https://github.com/babel/babel-preset-env/issues/54 example of edge case.
+
+### Help with Babel v7 Config
+
+It can be pretty painful to compile `node_modules` in Babel v6. The reason is both that we never thought of it as something we wanted as well as getting issue reports that people would actually accidently do it making the build slower.
+
+One issue (which we have fixed in v7) is around config lookup. Babel currently runs per file so when compiling a file, it tries to find the closest config to know what to compile against. If it doesn't find it in the same directory, it keeps looking up directories (this is why if you don't specify a config but have one in your home directory it might try to load that instead).
+
+```
+project
+└── .babelrc // closest config for a.js
+└── a.js
+└── node_modules
+    └── package
+        └── .babelrc // closest config for b.js
+        └── b.js
+```
+
+We made a [few changes](https://github.com/babel/babel/pull/7358):
+
+- One is to stop lookup at the package boundary (stop when you find a `package.json`). This makes sure Babel won't try to load a config file outside.
+- If you use a monorepo, you'll may want to have a `.babelrc` per-package that extends some other central config.
+- Babel itself is a monorepo, so instead we are using the new `babel.config.js` which allows you to resolve all files to that config (no more lookup).
+- We added an [`"overrides"`](https://github.com/babel/babel/pull/7091) option which allows you to basically create a new config for any set of paths.
+
+This allows you to have a single config for your whole app: maybe you want to compile your server JavaScript code differently than the client code as well as compile something in `node_modules`.
+
+```js
+// babel.config.js
+module.exports = {
+  presets: [
+    ['env', { 
+      targets: { node: 'current' },
+    }],
+  ],
+  overrides: [{
+    test: "./client",
+    presets: [
+      ['env', { 
+        targets: { browsers: ["last 2 versions"] },
+      }],
+    ],
+  }, {
+    test: "./node_modules/",
+    presets: [
+      ['env', { 
+        targets: { browsers: ["last 2 versions"] },
+      }],
+    ],
+  }],
+}
+```
+
+### Let's Do This!
+
+Hopefully this is an encouraging call to action for looking into moving forward to make compiling dependencies more first class. It's not just about the specific ES6/ES5 distinction.
+
+Babel v7 should be out soon, this post goes into some of the ways it should help with this effort but we'll need everyone's help to change the ecosystem (more education, published packages that do this, and better tooling and sustainability).
```

Comment:
> Babel v7 makes some headway in making compilation easier, but it's not enough to change only the tool. We'll need everyone's help...

Reply:
This should be your last paragraph.

Reply:
100% agree. Community/external tooling/platform support is going to be really necessary to pull this off, so the call to action based on B7’s work looks like it’s the best place to hand off the charge to the reader.